Frontend/ui-customer/src/Helpers/Utils.js

```diff
@@ -433,4 +433,214 @@ export const getMobileNumber = function (signatureAuthData, maskedNumber) {
 export const extractKeyByRegex = (regex, text) => {
   const matches = text.match(regex);
   return matches ? matches[0] : "";
-}
\ No newline at end of file
+}
+export const setSuggestionsMapInJson = function (map) {
+  console.log("zxc ",map);
+  let jsonString = JSON.stringify(map);
+  let unstring = JSON.parse(jsonString);
+  window.JBridge.setKeysInSharedPrefs("SUGGESTIONS_MAP", jsonString);
+  return Array.from(unstring);
+}
+
+export const getSuggestedDestinationsJsonFromLocal = function (key){
+  
+  let stringifiedMap=window.JBridge.getKeysInSharedPref ? window.JBridge.getKeysInSharedPref(key) : window.JBridge.getKeysInSharedPrefs(key); //  window.JBridge.getKeysInSharedPrefs(key);
+
+            if (stringifiedMap != "__failed" && stringifiedMap != "(null)") {
+                let unstring = JSON.parse(stringifiedMap);
+
+                console.log("parsed ", unstring)
+
+                return Array.from(unstring);
+            } 
+            console.log("err", stringifiedMap)
+            return stringifiedMap;
+}
+
+export const getDifferenceBetweenDates = function (date1) {
+  return function (date2) {
+    const diffInSeconds = Math.floor((new Date(date1) - new Date(date2)) / 1000);
+    return diffInSeconds;
+  }
+}
+
+export const parseSourceHashArray = function (String) {
+  console.log("parseSourceHashArray");
+  return JSON.parse(String);
+}
+
+/* --------------------- GEOHASH FUNCTIONS ---------------------- */
+
+const base32 = '0123456789bcdefghjkmnpqrstuvwxyz'; // (geohash-specific) Base32 map
+
+export const encodeGeohash = function (lat) {
+  return function (lon){  
+    return function(precision ){
+      if (typeof precision == 'undefined') {
+        for (let p = 1; p <= 12; p++) {
+          const hash = encodeGeohash(lat, lon, p);
+          const posn = decodeGeohash(hash);
+          if (posn.lat == lat && posn.lon == lon) return hash;
+        }
+        precision = 12; // set to maximum
+      }
+
+      lat = Number(lat);
+      lon = Number(lon);
+      precision = Number(precision);
+
+      if (isNaN(lat) || isNaN(lon) || isNaN(precision)) throw new Error('Invalid geohash');
```

Comment:
I think we need to return from this function after throwing this error

Reply:
`throw` will terminate the function here itself